modules/user_access/main.tf

```diff
@@ -0,0 +1,81 @@
+# User and Roles
+
+# Create group with polcy for AWS resource access
+resource "aws_iam_group" "access_group" {
+  count = length(var.roles)
+  name  = "${var.project}-${var.roles[count.index].name}-${var.environment}"
+  path  = "/users/"
+}
+
+resource "aws_iam_policy" "access_group" {
+  count       = length(var.roles)
+  name        = aws_iam_group.access_group[count.index].name
+  description = "Group policy"
+  policy      = var.roles[count.index].aws_policy
+}
+
+resource "aws_iam_group_policy_attachment" "access_group" {
+  count      = length(var.roles)
+  group      = aws_iam_group.access_group[count.index].name
+  policy_arn = aws_iam_policy.access_group[count.index].arn
+}
+
```

Comment:
Instead of having the assumerole policy included in the policy that is passed in, we should be creating it here. (this code from zero-aws-eks-stack:) 
```
  statement {
    effect = "Allow"
    actions = [
      "iam:ListRoles",
      "sts:AssumeRole"
    ]
    resources = ["arn:aws:iam::<% index .Params `accountId` %>:role/<% .Name %>-kubernetes-developer-prod"]
  }

```
It shouldn't be up to the user to have to include that policy block, we can create it dynamically here.

Reply:
done automatically in module side